src/store/modules/orders/actions.ts

```diff
@@ -0,0 +1,192 @@
+  },
+
+  async packDeliveryItems ({ commit }, shipmentId) {
+    const params = {
+      shipmentId: shipmentId,
+      statusId: 'SHIPMENT_PACKED'
+    }
+    return await OrderService.updateShipment(params)
+  },
+
+  async quickShipEntireShipGroup ({ dispatch }, payload) {
+    emitter.emit("presentLoader")
+
+    const params = {
+      orderId: payload.order.orderId,
+      setPackedOnly: 'Y',
+      dimensionUomId: 'WT_kg',
+      shipmentBoxTypeId: 'YOURPACKNG',
+      weight: '1',
+      weightUomId: 'WT_kg',
+      facilityId: payload.facilityId,
+      shipGroupSeqId: payload.shipGroup
+    }
+    
+    let resp;
+
+    try {
+      resp = await OrderService.quickShipEntireShipGroup(params)
+      if (resp.status === 200 && !hasError(resp) && resp.data._EVENT_MESSAGE_) {
+        const shipmentMethod = await dispatch('getShipmentMethod', {shipGroupSeqId: payload.shipGroup, items: payload.order.items})
+        if (shipmentMethod !== 'STOREPICKUP') {
+          const shipmentId = resp.data._EVENT_MESSAGE_.match(/\d+/g)[0]
+          await dispatch('packDeliveryItems', shipmentId).then((data) => {
+            if (!hasError(data) && !data.data._EVENT_MESSAGE_) showToast(translate("Something went wrong"))
+          })
+        }
+        emitter.emit("dismissLoader");
+        showToast(translate("Order packed and ready for delivery"))
+      } else {
+        emitter.emit("dismissLoader");
+        showToast(translate("Something went wrong"))
+      }
+    } catch(err) {
+      console.log(err)
+      emitter.emit("dismissLoader");
+      showToast(translate("Something went wrong"))
+    }
+
+    return resp;
+  },
+
+  async unfillableOrderOrItem ({ dispatch }, order) {
+    emitter.emit("presentLoader");
+    await dispatch("rejectOrderItems", order).then((resp) => {
+      let unfillableItems = 0;
+      const refreshPickupOrders = resp.find((response: any) => !(response.data._ERROR_MESSAGE_ || response.data._ERROR_MESSAGE_LIST_))
+      if (refreshPickupOrders) {
+        unfillableItems++;
+        showToast(`${unfillableItems} ${unfillableItems == 1 ? translate('item was') : translate('items were')}` + ' ' + translate('canceled from the order') + ' ' + order.orderId);
+        router.push('/tabs/orders')
+      } else {
+        showToast(translate('Something went wrong'));
+      }
+      emitter.emit("dismissLoader");
+    })
+  },
+
+  rejectOrderItems ({ commit }, order) {
+    const params: any = {
+      "payload": {
+        "orderId": order.orderId,
+        "rejectReason": ''
+      }
+    }
+
+    return Promise.all(order.items.map((item: any) => {
+      params['payload']['facilityId'] = item.facilityId
+      params['payload']['orderItemSeqId'] = item.orderItemSeqId
+      params['payload']['shipmentMethodTypeId'] = item.shipmentMethodTypeId
+      params['payload']['quantity'] = parseInt(item.inventory[0].quantity)
+      return OrderService.rejectOrderItem(params).catch((err) => { 
```

Comment:
I think there may be side effects. For multiple items, as we are not cloning the object but using the same reference there may be a chance payload is wrong and same for 2 subsequent ones

Reply:
Improved code to handle the case to properly prepare the payload object.